### PR TITLE
boards: spresense: enable lte modem

### PR DESCRIPTION
**Pull request by jerpelea**
## Summary
Enable and configure ALT1250 LTE modem

## Impact
SPRESENSE LTE board

## Testing
spresense

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: boards/arm/cxd56xx/common/src/cxd56_alt1250.c [hunk 1]
```diff
@@ -171,275 +167,262 @@ static void spi_pincontrol(int bus, bool on)
...
 /****************************************************************************
- * Name: altmdm_poweron
+ * Name: set_spiparam
+ *
+ * Description:
+ *   Setup the SPI parameters.
+ *
+ ****************************************************************************/
+
+static void set_spiparam(FAR struct spi_dev_s *spidev)
```
Comment (pkarashchenko) [inline boards/arm/cxd56xx/common/src/cxd56_alt1250.c:177]:
```suggestion
static void set_spiparam(struct spi_dev_s *spidev)
```

File: boards/arm/cxd56xx/common/src/cxd56_alt1250.c [hunk 2]
```diff
@@ -75,36 +73,34 @@
...
 
-static void altmdm_poweron(void);
-static void altmdm_poweroff(void);
-static void altmdm_sready_irqattach(bool attach, xcpt_t handler);
-static void altmdm_sready_irqenable(bool enable);
-static bool altmdm_sready(void);
-static void altmdm_master_request(bool request);
-static void altmdm_wakeup(bool wakeup);
-static uint32_t altmdm_spi_maxfreq(void);
+static FAR struct spi_dev_s *alt1250_poweron(void);
```
Comment (pkarashchenko) [inline boards/arm/cxd56xx/common/src/cxd56_alt1250.c:80]: Remove FAR in this file

File: boards/arm/cxd56xx/spresense/include/cxd56_alt1250.h [hunk 3]
```diff
@@ -46,47 +46,57 @@ extern "C"
...
- * Name: board_altmdm_initialize
+ * Name: board_alt1250_initialize
  *
  * Description:
  *   Initialize Altair modem
  *
  ****************************************************************************/
 
-int board_altmdm_initialize(const char *devpath);
+int board_alt1250_initialize(FAR const char *devpath);
```
Comment (SPRESENSE) [inline boards/arm/cxd56xx/spresense/include/cxd56_alt1250.h:59]:
```suggestion
int board_alt1250_initialize(const char *devpath);
```

Comment (jerpelea) [reply boards/arm/cxd56xx/spresense/include/cxd56_alt1250.h:59]: thanks for catching it

**Pull request merged**